src/drive/web/modules/actions/useActions.spec.jsx

```diff
@@ -0,0 +1,422 @@
+        false
+      )
+      expect(
+        trashActionWithoutWriteAccess.displayCondition(['abc', 'def'])
+      ).toBe(false)
+
+      mockSharingContextValue.hasWriteAccess.mockReturnValue(true)
+      const trashAction = getAction('trash')
+      expect(trashAction.displayCondition(['abc'])).toBe(true)
+      expect(trashAction.displayCondition(['abc', 'def'])).toBe(true)
+    })
+
+    it('shows the trash confirmation modal when activated', () => {
+      const trashAction = getAction('trash')
+      const mockDocuments = [
+        { id: 'abc', name: 'my-file.md' },
+        { id: 'def', name: 'my-file-2.md' }
+      ]
+      trashAction.action(mockDocuments)
+      const actuallyCalledModal =
+        mockModalContextValue.pushModal.mock.calls[0][0]
+      expect(mockModalContextValue.pushModal).toHaveBeenCalledWith(
+        <DeleteConfirm
+          files={mockDocuments}
+          onConfirm={actuallyCalledModal.props.onConfirm} // needs exact comparison
+        />
+      )
+    })
+  })
+
+  describe('open action', () => {
+    it('is only displayed on mobile with a single file selected', () => {
+      const openAction = getAction('openWith')
+
+      const validSelection = [{ type: 'file', id: 'abc' }]
+
+      global.__TARGET__ = 'mobile'
+      expect(openAction.displayCondition(validSelection)).toBe(true)
+
+      global.__TARGET__ = 'desktop'
+      expect(openAction.displayCondition(validSelection)).toBe(false)
+      expect(
+        openAction.displayCondition([
+          { type: 'file', id: 'abc' },
+          { type: 'file', id: 'def' }
+        ])
+      ).toBe(false)
+      expect(openAction.displayCondition([{ type: 'folder', id: 'abc' }])).toBe(
+        false
+      )
+    })
+
+    it('opens a file when activated', () => {
+      const openAction = getAction('openWith')
+      const mockDocument = { id: 'abc', name: 'my-file.md' }
+
+      openAction.action([mockDocument])
+      expect(openFileWith).toHaveBeenCalledWith(mockDocument, mockClient)
+    })
+  })
+
+  describe('rename action', () => {
+    it('is only visible with write access and a single selected item', () => {
+      mockSharingContextValue.hasWriteAccess.mockReturnValue(false)
+      const renameActionWithoutWriteAccess = getAction('rename')
+      expect(renameActionWithoutWriteAccess.displayCondition(['abc'])).toBe(
+        false
+      )
+      expect(
+        renameActionWithoutWriteAccess.displayCondition(['abc', 'def'])
+      ).toBe(false)
+
+      mockSharingContextValue.hasWriteAccess.mockReturnValue(true)
+      const renameAction = getAction('rename')
+      expect(renameAction.displayCondition(['abc'])).toBe(true)
+      expect(renameAction.displayCondition(['abc', 'def'])).toBe(false)
+    })
+
+    it('dispatches a rename action when activated', async () => {
+      const renameAction = getAction('rename')
+      const spy = jest.spyOn(rename, 'startRenamingAsync')
```

Comment:
Ideally I would have liked to test the call to `dispatch`, but somehow I can't get a reference to the `dispatch` returned by `useDispatch`. Even things like `const spy = jest.spyOn(mockStore, 'dispatch')` don't get called..?

Reply:
startRenamingAsync is a bit more down the line that dispatch right ? I think I prefer to have it this way, since we could think of `dispatch` as an implementation detail ?

Reply:
It is further down the line, but it also is an action creator so in a way it's couple to `dispatch`. Though we could change the implementation of `startRenamingAsync` without breaking the test too.